MeetupPlatform.Api/Features/Meetups/GetMeetups/GetMeetupsFeature.cs

```diff
@@ -24,12 +26,20 @@ public GetMeetupsFeature(ApplicationContext context, IMapper mapper)
     /// <response code="200">Returns meetup items colection.</response>
     [HttpGet("/api/meetups")]
     [ProducesResponseType(typeof(IEnumerable<MeetupInfoDto>), StatusCodes.Status200OK)]
-    public async Task<IActionResult> GetMeetups()
+    public async Task<IActionResult> GetMeetups([FromQuery] MeetupsFilterSettings filterSettings)
     {
-        var meetups = await context.Meetups
-            .Include(meetup => meetup.SignedUpUsers)
-            .ToListAsync();
-        var meetupInfoDtos = mapper.Map<IEnumerable<MeetupInfoDto>>(meetups);
-        return Ok(meetupInfoDtos);
+        try
+        {
+            var meetupsQuery = context.Meetups
+                .Include(meetup => meetup.SignedUpUsers)
+                    .GetMeetupsFilteredByFilterSettings(filterSettings);
+
+            var meetupInfoDtos = await mapper.ProjectTo<MeetupInfoDto>(meetupsQuery).ToListAsync();
+            return Ok(meetupInfoDtos);
+        }
+        catch (ArgumentOutOfRangeException)
```

Comment:
You don't need to use try-catch here. Instead You should validate provided value using validation attributes.